readthedocs/projects/querysets.py

```diff
@@ -131,27 +131,31 @@ def prefetch_latest_build(self):
         """
         from readthedocs.builds.models import Build
 
-        # Prefetch the latest build for each project.
-        subquery_build_latest = Subquery(
-            Build.internal.filter(project=OuterRef("project_id"))
-            .order_by("-date")
-            .values_list("id", flat=True)[:1]
+        # Get most recent and recent successful builds
+        builds_latest = (
+            Build.internal.filter(project__in=self)
```

Comment:
It's not clear why `Build.internal` is needed in both the inner query and the prefetch query. It seems like one of them could be `Build.objects` at very least? The performance is a little better with `Build.objects`.

Reply:
I understand is to avoid getting builds from PRs (external versions) and I'd say that `.internal` _should_ perform better than `.objects` since it removes _a lot_ of builds to consider and they should be removed using an index `Build.type`. That's the theory, tho 😄

Reply:
Heh same, that was my thought initially. There is some complexity added by this method though, which I think ultimately annoys the query planner.